src/products/views/ProductUpdate/handlers.ts

```diff
@@ -206,6 +254,34 @@ export function createUpdateHandler(
       }
     }
 
+    if (errors.length === 0) {
+      const deleteAttributeValuesResult = await Promise.all(
+        product.attributes.map(existingAttribute => {
+          const fileValueUnused =
+            existingAttribute.attribute.inputType ===
+              AttributeInputTypeEnum.FILE &&
+            existingAttribute.values.length > 0 &&
+            data.attributes.find(
+              dataAttribute =>
+                dataAttribute.id === existingAttribute.attribute.id
+            ).value.length === 0;
+
+          if (fileValueUnused) {
+            return deleteAttributeValue({
+              id: existingAttribute.values[0].id
+            });
+          }
+        })
+      );
+
+      deleteAttributeValuesResult.forEach(deleteValueResult => {
+        errors = [
+          ...errors,
+          ...deleteValueResult.data.attributeValueDelete.errors
+        ];
+      });
+    }
```

Comment:
Wouldn't `map` look better here?

Reply:
Not really, it merges one error array with many consecutive error arrays, so map can't help.

Reply:
`reduce` then?

Reply:
might be, updated